storage/database/src/error.rs

```diff
@@ -88,6 +88,10 @@ pub enum RuntimeError {
     #[error("database memory map must be resized")]
     ResizeNeeded,
 
+    /// The given table did not exist in the database.
+    #[error("database table did not exist")]
+    TableNotFound,
```

Comment:
Since `cuprate_database` doesn't know all the user's tables anymore, this runtime error now exists.

Although, `cuprate_blockchain::open` creates all the tables so this error never occurs.

Reply:
Also, `cuprate_database` no longer panics if a table doesn't exist, it returns this error.

Reply:
Actually... after your review I'm going to add a test for this.

Reply:
Is this planned for this PR?

Reply:
Oops, yes, but actually no. These tests already exist: https://github.com/hinto-janai/cuprate/blob/1648409727b00ee1032b866f4359fa4aea678e7a/storage/database/src/backend/tests.rs#L70-L116.

I added more tests to `cuprate_blockchain` though.